File: server/src/main/java/com/edutech/progressive/entity/Appointment.java
package com.edutech.progressive.entity;

import javax.persistence.*;
import java.util.Date;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "appointment")
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "appointment_id")
    private int appointmentId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Patient patient;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "clinic_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "clinics"})
    private Clinic clinic;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "appointment_date", nullable = false)
    private Date appointmentDate;

    @Column(name = "status", nullable = false, length = 255)
    private String status;

    @Column(name = "purpose", length = 255)
    private String purpose;

    public Appointment() {}

    public Appointment(int appointmentId, Patient patient, Clinic clinic, Date appointmentDate, String status, String purpose) {
        this.appointmentId = appointmentId;
        this.patient = patient;
        this.clinic = clinic;
        this.appointmentDate = appointmentDate;
        this.status = status;
        this.purpose = purpose;
    }

    public int getAppointmentId() {
        return appointmentId;
    }

    public void setAppointmentId(int appointmentId) {
        this.appointmentId = appointmentId;
    }

    public Patient getPatient() {
        return patient;
    }

    public void setPatient(Patient patient) {
        this.patient = patient;
    }

    public Clinic getClinic() {
        return clinic;
    }

    public void setClinic(Clinic clinic) {
        this.clinic = clinic;
    }

    public Date getAppointmentDate() {
        return appointmentDate;
    }

    public void setAppointmentDate(Date appointmentDate) {
        this.appointmentDate = appointmentDate;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getPurpose() {
        return purpose;
    }

    public void setPurpose(String purpose) {
        this.purpose = purpose;
    }
}

File: client/src/app/mediconnect/models/Appointment.ts
export class Appointment {
    appointmentId: number;
    patientId: number;
    clinicId: number;
    appointmentDate: Date;
    status: string;
    purpose: string;
  
    constructor(
      appointmentId: number,
      patientId: number,
      clinicId: number,
      appointmentDate: Date,
      status: string,
      purpose: string
    ) {
      this.appointmentId = appointmentId;
      this.clinicId = clinicId;
      this.patientId = patientId;
      this.appointmentDate = appointmentDate;
      this.status = status;
      this.purpose = purpose;
    }
  
    logAttributes(): void {
      console.log('appointmentId:', this.appointmentId);
      console.log('patientId:', this.patientId);
      console.log('clinicId:', this.clinicId);
      console.log('appointmentDate:', this.appointmentDate);
      console.log('status:', this.status);
      console.log('purpose:', this.purpose);
    }
  }
  